### PR TITLE
Improve integration tests framework

### PR DESCRIPTION
**Pull request by dongboyan77**
### Description of Changes

* Improve integration tests framework, such as: add env variables to enable headless, update packages, update README

#### Issue: <https://issues.redhat.com/browse/QUAY-2238>

---

## Reviewer Checklist

- [ ] It works!
- [ ] Comments provide sufficient explanations for the next contributor
- [ ] Tests cover changes and corner cases
- [ ] Follows Quay syntax patterns and format

### CODE REVIEW

**Comment by dongboyan77** [comment]
@thomasmckay thanks
@josephschorr @alecmerdler hi, could you help merge this PR, thanks

File: integration_tests/package.json [hunk 1]
```diff
@@ -2,24 +2,26 @@
   "description": "Quay Integration Testing",
```
Comment (alecmerdler) [inline integration_tests/package.json]: We should really actually move all this into the root `package.json`, but that can be another PR.

File: integration_tests/views/image-repository.view.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { $, browser, ExpectedConditions as until, by, element } from 'protractor';
+import { appHost, testName, userName } from '../protractor.conf';
+
+export const createrepoButton = element(by.linkText('Create New Repository'));
```
Comment (alecmerdler) [inline integration_tests/views/image-repository.view.ts]: Ensure proper `camelCase`.

File: integration_tests/tests/image-repository.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  });
+});
+
+describe('Viewing and modifying tags', () => {
+  beforeAll(async() => {
+    await browser.get(`${appHost}/repository/${userName}/${testName}?tab=tags`);
+    await browser.sleep(1000);
+  });
+
+/*  it('add new tag for image manifest', async() => {
```
Comment (alecmerdler) [inline integration_tests/tests/image-repository.scenario.ts]: We shouldn't have commented-out code. Is there something wrong with these tests?

File: integration_tests/protractor.conf.ts [hunk 4]
```diff
@@ -54,12 +56,12 @@ export const config: Config = {
         'password_manager_enabled': false,
       },
     },
-/**
- * 'browserName': 'firefox',
- * 'moz:firefoxOptions': {
- *   'args': ['--safe-mode']
- *   }
- */
+/*
```
Comment (alecmerdler) [inline integration_tests/protractor.conf.ts]: Shouldn't have commented-out code. I think it's safe to not expect Firefox testing for now.

File: integration_tests/package.json (hunk 1 above)
Comment (dongboyan77) [reply integration_tests/package.json]: Sure, it is OK for me, I will create another PR later

File: integration_tests/tests/image-repository.scenario.ts (hunk 3 above)
Comment (dongboyan77) [reply integration_tests/tests/image-repository.scenario.ts]: Because it is unstable due to locate elements by xpath, I want to improve this part in the future

File: integration_tests/protractor.conf.ts (hunk 4 above)
Comment (dongboyan77) [reply integration_tests/protractor.conf.ts]: OK, I will remove firefox

File: integration_tests/tests/image-repository.scenario.ts (hunk 3 above)
Comment (alecmerdler) [reply integration_tests/tests/image-repository.scenario.ts]: You can skip these tests [using `xit(...)`](https://jasmine.github.io/api/3.5/global.html#xit) instead of commenting out the code.

Comment (alecmerdler) [reply integration_tests/tests/image-repository.scenario.ts]: Please use `xit()` instead of commented-out code.

File: integration_tests/views/image-repository.view.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    await publicrepoButton.click();
+  } else {
+    await privaterepoClick.click();
+    await browser.wait(until.presenceOf(privaterepoButton));
+    await privaterepoButton.click();
+  }
+};
+
+export const addnewtag = async(previoustag: string, newtag: string) => {
+//  const optionsDropdown = element(by.xpath("//span[@bo-text=tag.name and text()='"+previoustag+"']/ancestor::tbody//i[@data-title=Options]"));
```
Comment (alecmerdler) [inline integration_tests/views/image-repository.view.ts]: Don't use commented-out code.

File: integration_tests/views/image-repository.view.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    await browser.wait(until.presenceOf(publicrepoButton));
+    await publicrepoButton.click();
+  } else {
+    await privaterepoClick.click();
+    await browser.wait(until.presenceOf(privaterepoButton));
+    await privaterepoButton.click();
+  }
+};
+
+export const addnewtag = async(previoustag: string, newtag: string) => {
```
Comment (alecmerdler) [inline integration_tests/views/image-repository.view.ts]: `addNewTag`

**Comment by dongboyan77** [comment]
Thanks @alecmerdler , I have moved this change to github.com/quay/quay-tests repo.
The issues of `camelCase`, skipping testing using `xlt` and commented-out code are modified.
Let me close the PR
